src/SendGrid/SendGridClient.cs

```diff
@@ -76,6 +82,35 @@ public SendGridClient(IWebProxy webProxy, string apiKey, string host = null, Dic
         /// <summary>
         /// Initializes a new instance of the <see cref="SendGridClient"/> class.
         /// </summary>
+        /// <param name="options">A <see cref="SendGridClientOptions"/> instance that defines the configuration settings to use with the client </param>
+        /// <returns>Interface to the SendGrid REST API</returns>
+        public SendGridClient(SendGridClientOptions options)
+            : this(null, options)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SendGridClient"/> class.
+        /// </summary>
+        /// <param name="httpClient">An optional http client which may me injected in order to facilitate testing.</param>
+        /// <param name="options">A <see cref="SendGridClientOptions"/> instance that defines the configuration settings to use with the client </param>
+        /// <returns>Interface to the SendGrid REST API</returns>
+        public SendGridClient(HttpClient httpClient, SendGridClientOptions options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            this.options = options;
+            client = (httpClient == null) ? CreateHttpClientWithRetryHandler() : httpClient;
```

Comment:
Yes, there are integration tests that use that and pass their own handler e.g. TestTakesHttpClientFactoryAsConstructorArgumentAndUsesItInHttpCalls. That preserves existing logic as per https://github.com/sendgrid/sendgrid-csharp/blob/f4092b7ac50353dbe51f0ac70f50ab69f1b61034/src/SendGrid/SendGridClient.cs#L88